src/coreclr/jit/gentree.cpp

```diff
@@ -7553,6 +7575,38 @@ GenTreeQmark* Compiler::gtNewQmarkNode(var_types type, GenTree* cond, GenTreeCol
     return result;
 }
 
+GenTreeRTCheck* Compiler::gtNewRTCheckNode(GenTree* value, GenTree* check, SpecialCodeKind kind)
+{
+    GenTreeRTCheck* node = new (this, GT_RTCHECK) GenTreeRTCheck(value, check, kind);
+    return node;
+}
+
+GenTreeRTCheck* Compiler::gtNewDivideByZeroCheck(GenTree* divisor)
+{
+    assert(varTypeIsIntegral(divisor));
+    GenTree* check = gtNewOperNode(GT_EQ, TYP_INT, gtCloneExpr(divisor), gtNewIconNode(0, genActualType(divisor)));
```

Comment:
It's not legal to clone the divisor tree like this since it can have side effects. I would just remove this and call `gtNewRTCheckNode` directly from the importer, where we can ensure the proper side effect handling.